### PR TITLE
feat(langgraph): add custom stream mode support in LangChain LLMAdapter

### PR DESCRIPTION
**Pull request by keenranger**
Langgraph not only supports messages streams but also [custom](https://docs.langchain.com/oss/python/langgraph/streaming#stream-custom-data). And it is only way to modify what to stream in livekit-langgraph integration.

So for that,
- Add `stream_mode` parameter supporting `"messages"` and `"custom"` modes
- Enable multi-mode streaming for LangGraph's StreamWriter output
- Extend `_to_chat_chunk()` to handle dict and object inputs

## Summary by CodeRabbit

* **New Features**
  * StreamMode configuration for LangGraph streaming with "messages" (default) and "custom" modes
  * Multi-mode streaming to mix and emit different payload types
  * Broader input acceptance: strings, dicts with content, and message-like objects; empty mode disables streaming

* **Tests**
  * Comprehensive tests for messages, custom, multi-mode streaming, edge cases and validation

* **Bug Fixes**
  * Validation now rejects unsupported stream modes and enforces allowed values

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: livekit-plugins/livekit-plugins-langchain/livekit/plugins/langchain/langgraph.py [hunk 1]
```diff
@@ -95,34 +109,63 @@ def __init__(
...
-            if token_like is None:
-                continue
+            # Multi-mode: item is (mode, data) tuple wrapper
+            if is_multi_mode and isinstance(item, tuple) and len(item) == 2:
+                mode, data = item
+                if isinstance(mode, str):
+                    if mode == "custom":
+                        # data = payload (str, dict, object)
+                        chat_chunk = _to_chat_chunk(data)
+                        if chat_chunk:
```
[the remark this answers is not in the document]
Comment (keenranger) [reply livekit-plugins/livekit-plugins-langchain/livekit/plugins/langchain/langgraph.py:148]: This was an existing issue before my PR and is not within the current scope.

**Comment by keenranger** [comment]
@davidzhao I saw you’ve reviewed related pr #3112 before.  
If you have time, could you take a look at this one as well? Thanks. :)

File: livekit-plugins/livekit-plugins-langchain/livekit/plugins/langchain/langgraph.py [hunk 2]
```diff
@@ -40,12 +43,20 @@ def __init__(
...
         subgraphs: bool = False,
+        stream_mode: StreamMode | list[StreamMode] = "messages",
     ) -> None:
         super().__init__()
+        modes = {stream_mode} if isinstance(stream_mode, str) else set(stream_mode)
+        unsupported = modes - _SUPPORTED_MODES
+        if unsupported:
+            raise ValueError(
+                f"Unsupported stream mode(s): {unsupported}. Only {_SUPPORTED_MODES} are supported."
+            )
```
[the remark this answers is not in the document]
Comment (keenranger) [reply livekit-plugins/livekit-plugins-langchain/livekit/plugins/langchain/langgraph.py:54]: if they don't want stream_mode, they can use empty stream_mode list

**Comment by keenranger** [comment]
Thanks for the review, @davidzhao.

- For `messages`, we can use `GenericFakeChatModel` to assert token streaming without calling a real LLM.
- For `custom`, a small LangGraph node with `StreamWriter` should be enough to verify payloads pass through unchanged.
- Then for multi-mode, we can combine both and check routing + ordering.

If that sounds reasonable, I can add a small pytest example or follow whatever testing style you prefer here.

**Comment by davidzhao** [comment]
@keenranger that sounds great!

**Comment by keenranger** [comment]
@davidzhao I made some test cases that fulfills requirements.

We have `messages` `custom` `both` case for stream_mode
And `LLM(messages)` `StreamWriter(custom)` `combined` for actual stream.

| Test | stream_mode | Graph | Verifies |
|------|-------------|-------|----------|
| `test_messages_mode` | `"messages"` | LLM | Token streaming works |
| `test_messages_mode_is_default` | (default) | LLM | Backward compatibility |
| `test_custom_mode_string` | `"custom"` | StreamWriter | String payloads work |
| `test_custom_mode_dict` | `"custom"` | StreamWriter | Dict `{"content":...}` works |
| `test_multi_mode` | `["messages","custom"]` | Both | Multi-mode routing |
| `test_empty_stream_mode_disables_streaming` | `[]` | Both | Opt-out of streaming |
| `test_custom_mode_no_messages_output` | `"custom"` | LLM | Mode isolation (no leak) |
| `test_messages_mode_no_custom_output` | `"messages"` | StreamWriter | Mode isolation (no leak) |
| `test_validation_rejects_unsupported_mode` | `"values"` | - | Invalid mode rejected |
| `test_validation_rejects_unsupported_in_list` | `["messages","updates"]` | - | Invalid in list rejected |
| `test_validation_accepts_supported_modes` | various | - | Valid modes accepted |
